src/background.c

```diff
@@ -111,6 +112,70 @@ get_permission (const char *app_id)
   return UNSET;
 }
 
+static GVariant *
+get_all_permissions (void)
+{
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GVariant) out_perms = NULL;
+  g_autoptr(GVariant) out_data = NULL;
+
+  if (!xdp_impl_permission_store_call_lookup_sync (get_permission_store (),
+                                                   PERMISSION_TABLE,
+                                                   PERMISSION_ID,
+                                                   &out_perms,
+                                                   &out_data,
+                                                   NULL,
+                                                   &error))
+    {
+      g_dbus_error_strip_remote_error (error);
+      g_debug ("No background permissions found: %s", error->message);
+      return NULL;
+    }
+
+  return g_steal_pointer (&out_perms);
+}
+
+static Permission
+get_one_permission (const char *app_id,
+                    GVariant   *perms)
```

Comment:
Can get_permission() not use this helper too?

Reply:
In fact, it could probably call get_all_permissions() first and then get_one_permission().